_posts/2018-08-01-gke-multi-stage.md

```diff
@@ -0,0 +1,424 @@
+---
+title: Multi-stage Serverless on Kubernetes with OpenFaaS and GKE
+description: A guide on setting up a cost effective, auto-scalable, multi-stage OpenFaaS on Google Cloud Kubernetes Engine
+date: 2018-08-01
+image: /images/gke-multi-stage/pexels-architecture-buildings-city-185662.jpg
+categories:
+  - auto-scaling
+  - kubernetes
+author_staff_member: stefan
+---
+
+This is a guide on setting up a cost effective, auto-scalable, multi-stage OpenFaaS on Google Cloud Kubernetes Engine.
+
+Experience level:
+
+intermediate OpenFaaS / intermediate GKE
+
+At the end of this guide you will be running two OpenFaaS environments on the same GKE cluster with the following characteristics: 
+* a dedicated GKE node pool for OpenFaaS core services 
+* a dedicated node pool made out of preemptible VMs for OpenFaaS functions 
+* autoscaling for functions and their underlying infrastructure
+* isolated staging and production environments with network policies
+* secure OpenFaaS ingress with Let's Encrypt TLS and authentication
+
+![openfaas-gke](/images/gke-multi-stage/overview.png)
+
+This setup can enable multiple teams to share the same CD pipeline with staging/production environments 
+hosted on GKE and development taking place on a local environment such as Minikube or Docker for Mac.
+
+## GKE cluster setup 
```

Comment:
We will be creating a cluster on Google's Kubernetes Engine (GKE), if you don't have an account you can sign up [here] for free credit.